userService/src/main/java/com/userServiceCrudOps/springbootcrudeUserService/service/UserProfileService.java

```diff
@@ -0,0 +1,60 @@
+package com.userServiceCrudOps.springbootcrudeUserService.service;
+
+import com.userServiceCrudOps.springbootcrudeUserService.entity.UserProfile;
+import com.userServiceCrudOps.springbootcrudeUserService.repository.UserProfileRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+@Service
+public class UserProfileService {
+
+    @Autowired
+    private UserProfileRepository repository;
```

Comment:
Constructor injection is expected. Class variables should preferably be initialized using a constructor.

Reply:
Autowired does the same thing as well i think.

Reply:
Constructor injection is considered one of the best practices in Spring Boot development due to its advantages in promoting immutability, explicit dependencies, mandatory initialization, easier unit testing, better compatibility with frameworks, and reduced complexity.